vlasovsolver/vlasovmover.cpp

```diff
@@ -303,31 +309,42 @@ void calculateSpatialTranslation(
          remoteTargetCellsx,
          remoteTargetCellsy,
          remoteTargetCellsz,
-         nPencils,
+         nPencilsX,
+         nPencilsY,
+         nPencilsZ,
          dt,
          popID,
          time
       );
    }
-   
+
    if (Parameters::prepareForRebalance == true) {
+      // Blocks need to be scaled to time since acceleration uses time
```

Comment:
Except that currently we don't actually include the time from acceleration in the weight.

And using time here is IMO a Bad Idea(tm) because the thing being timed is calculateSpatialTranslation(), which includes all the MPI barriers, ghost updates, remote neighbour contributions etc.

Reply:
I think you are right, I can revert that crap. In any case it's just a matter of rescaling so shouldn't affect results.